tests/unit/data_tests.py

```diff
@@ -11,42 +11,51 @@
 except ImportError:
     import unittest
 
+try:
+    from collections import OrderedDict
+except ImportError:
+    from ordereddict import OrderedDict
+
 from pika import data
 from pika import exceptions
+from pika.compat import long
 
 
 class DataTests(unittest.TestCase):
 
-    FIELD_TBL_ENCODED = ('\x00\x00\x00\xbb\x07longvall\x00\x00\x00\x006e&U'
-                         '\x06intvalI\x00\x00\x00\x01\x07dictvalF\x00\x00'
-                         '\x00\x0c\x03fooS\x00\x00\x00\x03bar\x07unicodeS'
-                         '\x00\x00\x00\x08utf8=\xe2\x9c\x93\x05arrayA\x00'
-                         '\x00\x00\x0fI\x00\x00\x00\x01I\x00\x00\x00\x02I'
-                         '\x00\x00\x00\x03\x04nullV\x06strvalS\x00\x00\x00'
-                         '\x04Test\x0ctimestampvalT\x00\x00\x00\x00Ec)\x92'
-                         '\x07decimalD\x02\x00\x00\x01:\x07boolvalt\x01'
-                         '\x0bdecimal_tooD\x00\x00\x00\x00d')
+    FIELD_TBL_ENCODED = (
+        b'\x00\x00\x00\xbb'
+        b'\x05arrayA\x00\x00\x00\x0fI\x00\x00\x00\x01I\x00\x00\x00\x02I\x00\x00\x00\x03'
+        b'\x07boolvalt\x01'
+        b'\x07decimalD\x02\x00\x00\x01:'
+        b'\x0bdecimal_tooD\x00\x00\x00\x00d'
+        b'\x07dictvalF\x00\x00\x00\x0c\x03fooS\x00\x00\x00\x03bar'
+        b'\x06intvalI\x00\x00\x00\x01'
+        b'\x07longvall\x00\x00\x00\x006e&U'
+        b'\x04nullV'
+        b'\x06strvalS\x00\x00\x00\x04Test'
+        b'\x0ctimestampvalT\x00\x00\x00\x00Ec)\x92'
+        b'\x07unicodeS\x00\x00\x00\x08utf8=\xe2\x9c\x93'
+    )
 
```

Comment:
The encoded table contents are reordered to match the order of insertion to the OrderedDict